src/lib/arch/aarch64/ExceptionHandler.cc

```diff
@@ -13,462 +10,98 @@ namespace simeng {
 namespace arch {
 namespace aarch64 {
 
-ExceptionHandler::ExceptionHandler(
-    const std::shared_ptr<simeng::Instruction>& instruction, const Core& core,
-    MemoryInterface& memory, std::shared_ptr<OS::SyscallHandler> sysHandler)
-    : instruction_(*static_cast<Instruction*>(instruction.get())),
-      core(core),
-      memory_(memory),
-      sysHandler_(sysHandler) {
-  resumeHandling_ = [this]() { return init(); };
+ExceptionHandler::ExceptionHandler(const Core& core) : core_(core) {}
+
+bool ExceptionHandler::tick() {
+  // If the syscall handler has been envoked, conclude the syscall only once the
```

Comment:
Would this better if reworded to say: If an exception corresponding to a syscall was encountered and passed to the simulated Operating System's syscall handler, conclude the syscall only once the result has been returned?

Reply:
Happy with this more explicit comment